genrl/deep/agents/dqn/prioritized.py

```diff
@@ -0,0 +1,76 @@
+import torch
+from torch import optim as opt
+
+from genrl.deep.agents.dqn.base import BaseDQN
+
+
+class PrioritizedReplayDQN(BaseDQN):
```

Comment:
Why is this a separate class altogether? What if someone wants to use `CategoricalDQN` with `PrioritizedReplayBuffer`?

Reply:
Yeah, so what I was actually thinking of was these are all model classes.

If someone wants to a combination of these, then they should be able to make their own custom DQN by adding in custom snippets of code. Like for example the `ddqn_q_target` or the `projection_distribution`. I can segment the code into even more functions so that it gets easier to use. I'll put an example up today. 

For specifically PrioritizedBuffer, i was thinking we can make that available for all OffPolicy algos to use (idk about SAC but DDPG and TD3 had slight variants allowing them to do so). It's not standard but it's an additional feature.